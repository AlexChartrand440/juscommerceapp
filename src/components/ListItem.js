import React from 'react';
import styled from 'styled-components';
import { Image } from 'react-native';
import { Card } from 'react-native-paper';

const CardItem = styled.TouchableOpacity`
  padding-vertical: 8;
`;

const Wrapper = styled.View`
  flex-direction: row;
`;

const DetailsView = styled.View`
  flex: 2;
  flex-direction: column;
  padding-top: 6;
`;

const TitleView = styled.View`
  flex-direction: column;
`;

const ItemTitle = styled.Text`
  flex: 4;
  font-size: 18;
  padding-left: 12;
  color: #202020;
`;

const ItemSubTitle = styled.Text`
  flex: 4;
  font-size: 12;
  padding-left: 12;
  color: #444444;
`;

const DetailsFooterView = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: flex-end;
  justify-content: space-between;
`;

const PriceText = styled.Text`
  flex: 1;
  color: #ff685f;
  font-size: 20;
  font-weight: 500;
  padding-left: 12;
  padding-bottom: 12;
`;

const ImageView = styled.View`
  flex: 1;
  align-items: flex-start;
`;

const ItemImage = styled(Image)`
  width: 120;
  height: 100;
  border-radius: 5;
`;

export default ({ item, onPress }) => (
  <CardItem onPress={onPress}>
    <Card>
      <Wrapper>
        <ImageView>
          <ItemImage source={{ uri: item.image }} />
        </ImageView>
        <DetailsView>
          <TitleView>
            <ItemTitle>{item.name}</ItemTitle>
            <ItemSubTitle>{item.description}</ItemSubTitle>
          </TitleView>
          <DetailsFooterView>
            <PriceText>{`R$ ${item.price}`}</PriceText>
          </DetailsFooterView>
        </DetailsView>
      </Wrapper>
    </Card>
  </CardItem>
);
