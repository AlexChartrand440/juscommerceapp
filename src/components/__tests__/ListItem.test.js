import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../ListItem';

const product = {
  id: 1,
  name: 'test product1',
  description: 'test description 1',
  image:
    'https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681',
  price: 123
};

test('test list item', () => {
  const wrapper = renderer.create(<ListItem item={product} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
