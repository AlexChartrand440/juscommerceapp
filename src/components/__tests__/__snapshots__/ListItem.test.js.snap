// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test list item 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View
        index={0}
        siblings={
          Array [
            "Styled(View)",
          ]
        }
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
        total={1}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2017/09/12/11/naturo-monkey-selfie.jpg?w968h681",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 5,
                  "height": 100,
                  "width": 120,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 2,
                "flexShrink": 1,
                "paddingTop": 6,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#202020",
                    "flexBasis": 0,
                    "flexGrow": 4,
                    "flexShrink": 1,
                    "fontSize": 18,
                    "paddingLeft": 12,
                  },
                ]
              }
            >
              test product1
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#444444",
                    "flexBasis": 0,
                    "flexGrow": 4,
                    "flexShrink": 1,
                    "fontSize": 12,
                    "paddingLeft": 12,
                  },
                ]
              }
            >
              test description 1
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ff685f",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 20,
                    "fontWeight": "500",
                    "paddingBottom": 12,
                    "paddingLeft": 12,
                  },
                ]
              }
            >
              R$ 123
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
